### PR TITLE
[MRG] Make _parse_bids_filename() public, drop verbose arg

### PR DESCRIPTION
**Pull request by hoechenberger**
PR Description
--------------

We make use of `mne_bids.path._parse_bids_filename()` in the mne-study-template in a totally un-hacky manner, making us believe that this function should probably be made public.

This PR also drops the function's `verbose` argument, as it was unused.

Describe your PR here

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"
- [ ] Commit history does not contain any merge commits

### CODE REVIEW

File: doc/whats_new.rst [hunk 1]
```diff
@@ -65,6 +65,7 @@ API
 - :func:`mne_bids.make_dataset_description` now takes the argument `overwrite` which will reset all fields if `True`. If `False`, user-provided fields will no longer be overwritten by :func:`mne_bids.write_raw_bids` when its `overwrite` argument is `True` unless new values are supplied, by `Alex Rockhill`_ (`#478 <https://github.com/mne-tools/mne-bids/pull/478>`_)
 - :func:`mne_bids.make_report` is now available from the `mne_bids` namespace that creates a string output of a summary of the BIDS dataset. In addition, the command line interface allows one to call `make_report`, by `Adam Li`_ (`#457 <https://github.com/mne-tools/mne-bids/pull/457>`_)
 - Added namespace :code:`mne_bids.path` which hosts path-like functionality for MNE-BIDS by `Adam Li`_ (`#483 <https://github.com/mne-tools/mne-bids/pull/483>`_)
+- :func:`mne_bids.path._parse_bids_filename` is now part of the public API and has consequently been renamed to `:func:`mne_bids.path._parse_bids_filename`, by `Richard Höchenberger`_ (`#487 <https://github.com/mne-tools/mne-bids/pull/487>`_)
```
Comment (agramfort) [inline doc/whats_new.rst:68]: private functions cannot be put in the what's new as they are not part of the public API

Comment (agramfort) [reply doc/whats_new.rst:68]: you also need to add to API rst page the new public function.

Comment (hoechenberger) [reply doc/whats_new.rst:68]: > private functions cannot be put in the what's new as they are not part of the public API

But it is now… so it's a "new" function. Do you mean I should not list it under "API", but instead under Changelog?



> you also need to add to API rst page the new public function.

will do

**Comment by agramfort** [comment]
you wrote _parse_bids_filename

>

**Comment by hoechenberger** [comment]
> you wrote _parse_bids_filename

I wrote:

> :func:`mne_bids.path._parse_bids_filename` is now part of the public API and has consequently been renamed to `:func:`mne_bids.path._parse_bids_filename`, 

But I think I see what you mean. Will rephrase.

**Pull request merged**